py/server/deephaven/parquet.py

```diff
@@ -280,6 +288,93 @@ def write(
         raise DHError(e, "failed to write to parquet data.") from e
 
 
+def write_key_value_partitioned_table(
+        table: Union[Table, PartitionedTable],
+        destination_dir: str,
+        col_definitions: Optional[List[Column]] = None,
+        col_instructions: Optional[List[ColumnInstruction]] = None,
+        compression_codec_name: Optional[str] = None,
+        max_dictionary_keys: Optional[int] = None,
+        max_dictionary_size: Optional[int] = None,
+        target_page_size: Optional[int] = None,
+        base_name: Optional[str] = None,
+        generate_metadata_files: Optional[bool] = False,
+) -> None:
+    """ Write a table to a Parquet file in a key-value partitoned format.
+
+    Args:
+        table (Table): the source table or partitioned table
+        destination_dir (str): the destination root directory to store partitioned data in nested format. Non-existing
+            directories are created.
+        col_definitions (Optional[List[Column]]): the column definitions to use, default is None
+        col_instructions (Optional[List[ColumnInstruction]]): instructions for customizations while writing, default is None
+        compression_codec_name (Optional[str]): the default compression codec to use, if not specified, defaults to SNAPPY
+        max_dictionary_keys (Optional[int]): the maximum dictionary keys allowed, if not specified, defaults to 2^20 (1,048,576)
+        max_dictionary_size (Optional[int]): the maximum dictionary size (in bytes) allowed, defaults to 2^20 (1,048,576)
+        target_page_size (Optional[int]): the target page size in bytes, if not specified, defaults to 2^20 bytes (1 MiB)
+        base_name (Optional[str]): The base name for the individual partitioned tables, if not specified, defaults to
+            `{uuid}`, so files will have names of the format `<uuid>.parquet` where `uuid` is a randomly generated UUID.
+            Users can provide the following tokens in the base_name:
+            - The token `{uuid}` will be replaced with a random UUID. For example, a base name of
+            "table-{uuid}" will result in files named like "table-8e8ab6b2-62f2-40d1-8191-1c5b70c5f330.parquet.parquet".
+            - The token `{partitions}` will be replaced with an underscore-delimited, concatenated string of
+            partition values. For example, a base name of "{partitions}-table" will result in files like
+            "PC1=partition1/PC2=partitionA/PC1=partition1_PC2=partitionA-table.parquet", where "PC1" and "PC2" are
+            partitioning columns.
```

Comment:
1. Does this case create subdirectories?
2. I'm confused by this.  Why are the partition values listed 2x?

Reply:
Looking more, it might make sense, but it will need clarification for users.